### PR TITLE
Issue 29- Stack.py

### PR DESCRIPTION
**Pull request by shirapahmer**
Stack.py is now dependent on Visualization App to run push and pop. Additonally the cell size is smaller to accomodate a stack of 8. There is also a pointer/index at the top of the stack.

### CODE REVIEW

File: PythonVisualizations/Stack.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    # STACK FUNCTIONALITY
+
+    def createIndex(self, index, name=None):
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        x0 = STACK_X0 - CELL_SIZE * 4 // 5
+        x1 = STACK_X0 - CELL_SIZE * 3 // 10
+        y0 = y1 = cell_coords[1] + CELL_SIZE // 2
+        return self.drawArrow(
+            x0, y0, x1, y1, color = VARIABLE_COLOR, font= VARIABLE_FONT)
```
Comment (JMCanning78) [inline PythonVisualizations/Stack.py:43]: The optional parameter `name` should be passed to the `drawArrow()` method in this call.

File: PythonVisualizations/Stack.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+CELL_SIZE = 40
+CELL_BORDER = 2
+CELL_BORDER_COLOR = 'black'
+STACK_X0 = 300
+STACK_Y0 =  400
+FONT_SIZE = 20
+VALUE_FONT = ('Helvetica', FONT_SIZE)
+VALUE_COLOR = 'black'
+
```
Comment (JMCanning78) [inline PythonVisualizations/Stack.py:18]: Please merge `origin/master` into this issue branch to change these global constants into class attributes.  The references in the methods to these constants will then need to be prefixed with `self.`

File: PythonVisualizations/Stack.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                x0 - CELL_SIZE*2 , y0+ CELL_SIZE/5, text=name, anchor=SW,
+                font=font, fill=color)
+        return (arrow, label) if name else (arrow, )
+
+    def push(self, val):
+        self.cleanUp()
+
+        # draw an index pointing to the last cell
+        indexDisplay = self.createIndex(len(self.list))
+        self.cleanup |= set(indexDisplay)
```
Comment (JMCanning78) [inline PythonVisualizations/Stack.py:60]: The set up of the call environment is different now.  Merging `origin/master` into this issue branch will enable you to change these initial lines of the `push()` method to:
```
    callEnviron = self.createCallEnvironment()
    indexDisplay = self.createIndex(len(self.list))
    callEnviron |= set(indexDisplay)
```

**Comment by shirapahmer** [comment]
this branch also is dealt with in pull request #63  and the changes are all found there so I don't think we need this one either

**Comment by JMCanning78** [comment]
We can close this (#58) when PR #49 and #59 are all handled by (#63).  For now, these old pull requests are here to remind us of all the changes that were requested at different times.  PR #63 must be evaluated to see that it covers all of them.

**Comment by JMCanning78** [comment]
Closed by PR #63

**Pull request closed without merging**